hack/cherry_pick_pull.sh

```diff
@@ -0,0 +1,109 @@
+# meta.) Assumes you care about pulls from remote "upstream" and
+# checks thems out to a branch named pull_12345.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare -r KUBE_ROOT="$(dirname "${BASH_SOURCE}")/.."
+cd "${KUBE_ROOT}"
+
+declare -r STARTINGBRANCH=$(git symbolic-ref --short HEAD)
+
+if [[ "$#" -ne 2 ]]; then
+  echo "${0} <pr-number> <branch>: cherry pick <pr> onto <branch> and leave instructions for proposing pull request"
+  exit 2
+fi
+
+if git_status=$(git status --porcelain 2>/dev/null) && [[ -n ${git_status} ]]; then
+  echo "!!! Dirty tree. Clean up and try again."
+  exit 1
+fi
+
+declare -r PULL="${1}"
+declare -r BRANCH="${2}"
+echo "+++ Updating remotes..."
+git remote update
+
+echo "+++ Downloading patch to /tmp/${PULL}.patch (in case you need to do this again)"
+
+curl -o "/tmp/${PULL}.patch" -sSL "https://github.com/GoogleCloudPlatform/kubernetes/pull/${PULL}.patch"
+
+declare -r NEWBRANCH="$(echo automated-cherry-pick-of-#${PULL}-on-${BRANCH} | sed 's/\//-/g')"
+declare -r NEWBRANCHUNIQ="${NEWBRANCH}-$(date +%s)"
+echo "+++ Creating local branch ${NEWBRANCHUNIQ}"
+
+function return_to_kansas {
+  git checkout -f "${STARTINGBRANCH}"
+}
+trap return_to_kansas EXIT
+
+git checkout -b "${NEWBRANCHUNIQ}" "${BRANCH}"
+
+echo
+echo "+++ About to attempt cherry pick of PR. To reattempt:"
+echo "  $ git am -3 /tmp/${PULL}.patch"
+echo
+git am -3 "/tmp/${PULL}.patch" || {
+  while unmerged=$(git status --porcelain | grep ^U) && [[ -n ${unmerged} ]]; do
+    echo
+    echo "+++ Conflicts detected:"
+    echo
+    git status --porcelain | grep ^U
+    echo
+    echo "+++ Please resolve the conflicts in another window (and remember to 'git add')"
+    read -p "+++ Proceed (anything but 'y' aborts the cherry-pick)? [y/n] " -r
+    echo
+    [[ $REPLY =~ ^[yY]$ ]] || {
+      echo "Aborting." >&2
+      exit 1
+    }
+  done
+}
+
+if git remote -v | grep ^origin | grep GoogleCloudPlatform/kubernetes.git; then
+  echo "!!! You have 'origin' configured as your GoogleCloudPlatform/kubernetes.git"
+  echo "This isn't normal. Leaving you with push instructions:"
+  echo
+  echo "  git push REMOTE ${NEWBRANCHUNIQ}:${NEWBRANCH}"
+  echo
+  echo "where REMOTE is your personal fork (maybe 'upstream'? Consider swapping those.)."
+  echo "Then propose ${NEWBRANCH} as a pull against ${BRANCH} (NOT MASTER)."
+  echo "Use this exact subject: 'Autogenerated cherry-pick of #${PULL}' and include a justification."
+  exit 1
+fi
+
+echo
+echo "+++ I'm about to do the following to push to GitHub (and I'm assuming origin is your personal fork):"
+echo
+echo "  git push origin ${NEWBRANCHUNIQ}:${NEWBRANCH}"
+echo
+read -p "+++ Proceed (anything but 'y' aborts the cherry-pick)? [y/n] " -r
+[[ $REPLY =~ ^[yY]$ ]] || {
+  echo "Aborting." >&2
+  exit 1
+}
+
+git push origin -f "${NEWBRANCHUNIQ}:${NEWBRANCH}"
+
+echo
+echo "+++ Now propose ${NEWBRANCH} as a pull against ${BRANCH} (NOT MASTER)."
```

Comment:
Since this is a tool, can you please clarify whether the action is being done automatically by the tool vs. the person running it?  E.g, something like "You must now XXX"?  Bonus points for constructing a github URL to do that action.